app/code/Magento/CompareListGraphQl/Model/Service/GetComparableItems.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Service;
+
+use Magento\Catalog\Block\Product\Compare\ListCompare;
+use Magento\Catalog\Model\CompareListIdToMaskedListId;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\ProductRepository;
+use Magento\CompareListGraphQl\Model\Service\Collection\GetComparableItemsCollection as ComparableItemsCollection;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Store\Api\Data\StoreInterface;
+
+/**
+ * Get comparable products
+ */
+class GetComparableItems
+{
+    /**
+     * @var ListCompare
+     */
+    private $blockListCompare;
+
+    /**
+     * @var ComparableItemsCollection
+     */
+    private $comparableItemsCollection;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var CompareListIdToMaskedListId
+     */
+    private $compareListIdToMaskedListId;
+
+    /**
+     * @param ListCompare $listCompare
+     * @param ComparableItemsCollection $comparableItemsCollection
+     * @param ProductRepository $productRepository
+     * @param CompareListIdToMaskedListId $compareListIdToMaskedListId
+     */
+    public function __construct(
+        ListCompare $listCompare,
+        ComparableItemsCollection $comparableItemsCollection,
+        ProductRepository $productRepository,
+        CompareListIdToMaskedListId $compareListIdToMaskedListId
+    ) {
+        $this->blockListCompare = $listCompare;
+        $this->comparableItemsCollection = $comparableItemsCollection;
+        $this->productRepository = $productRepository;
+        $this->compareListIdToMaskedListId = $compareListIdToMaskedListId;
+    }
+
+    /**
+     * Get comparable items
+     *
+     * @param int $listId
+     * @param ContextInterface $context
+     *
+     * @return array
+     * @throws GraphQlInputException
+     */
+    public function getComparableItems(int $listId, ContextInterface $context)
```

Comment:
Please, rename it to `execute()`. We already have a meaningful name of the class itself.

Reply:
**Done**